<ui version="4.0" >
 <class>DebuggerConfigWidget</class>
 <widget class="QWidget" name="DebuggerConfigWidget" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>578</width>
    <height>638</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>Debugger Configuration</string>
  </property>
  <property name="toolTip" >
   <string/>
  </property>
  <property name="whatsThis" >
   <string/>
  </property>
  <layout class="QGridLayout" >
   <item row="0" column="0" >
    <widget class="QLabel" name="textLabel1" >
     <property name="text" >
      <string>Debug arguments can be set on the Run Options page
 or directly in the project manager</string>
     </property>
     <property name="wordWrap" >
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="1" column="0" >
    <widget class="QLabel" name="gdbPath_label" >
     <property name="sizePolicy" >
      <sizepolicy vsizetype="Preferred" hsizetype="Maximum" >
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text" >
      <string>Debugger executable:</string>
     </property>
     <property name="wordWrap" >
      <bool>false</bool>
     </property>
     <property name="buddy" >
      <cstring>gdbPath_edit</cstring>
     </property>
    </widget>
   </item>
   <item row="2" column="0" >
    <widget class="KUrlRequester" name="gdbPath_edit" >
     <property name="toolTip" >
      <string>Gdb executable</string>
     </property>
     <property name="whatsThis" >
      <string>To run "gdb" binary from $PATH, leave this field empty. To run custom gdb, for example, for a different architecture, enter the executable name here. You can either run gdb from $PATH, but with a different name (say, "arm-gdb"), by typing the name here, or specify full path to the gdb executable.</string>
     </property>
    </widget>
   </item>
   <item row="3" column="0" >
    <widget class="QLabel" name="debuggingShell_label" >
     <property name="sizePolicy" >
      <sizepolicy vsizetype="Preferred" hsizetype="Maximum" >
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="toolTip" >
      <string>Run gdb in a special shell (mainly for automake projects)</string>
     </property>
     <property name="whatsThis" >
      <string>If you want gdb to be executed by a special shell or tool insert it here. The main use-case is for Automake based projects where the application is actually only a script and libtool is needed to get it to run inside gdb.</string>
     </property>
     <property name="text" >
      <string>Debugging &amp;shell:</string>
     </property>
     <property name="wordWrap" >
      <bool>false</bool>
     </property>
     <property name="buddy" >
      <cstring>debuggingShell_edit</cstring>
     </property>
    </widget>
   </item>
   <item row="4" column="0" >
    <widget class="KUrlRequester" name="debuggingShell_edit" >
     <property name="toolTip" >
      <string>Run gdb in a special shell (mainly for automake projects)</string>
     </property>
     <property name="whatsThis" >
      <string>If you want gdb to be executed by a special shell or tool insert it here. The main use-case is for Automake based projects where the application is actually only a script and &lt;b>libtool&lt;/b> is needed to get it to run inside gdb.</string>
     </property>
    </widget>
   </item>
   <item row="5" column="0" >
    <layout class="QGridLayout" >
     <item rowspan="2" row="0" column="0" >
      <widget class="QGroupBox" name="groupBox_2" >
       <property name="title" >
        <string>Options</string>
       </property>
       <layout class="QVBoxLayout" >
        <item>
         <widget class="QCheckBox" name="displayStaticMembers_box" >
          <property name="whatsThis" >
           <string>Displaying static members makes GDB slower in
producing data within KDE and Qt.
It may change the "signature" of the data
which QString and friends rely on,
but if you need to debug into these values then
check this option.</string>
          </property>
          <property name="text" >
           <string>Display static &amp;members</string>
          </property>
          <property name="shortcut" >
           <string>Alt+M</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="asmDemangle_box" >
          <property name="whatsThis" >
           <string>When displaying the disassembled code you
can select to see the methods' mangled names.
However, non-mangled names are easier to read.</string>
          </property>
          <property name="text" >
           <string>Display &amp;demangled names</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="breakOnLoadingLibrary_box" >
          <property name="whatsThis" >
           <string>If GDB has not seen a library that will be loaded via
"dlopen" then it will refuse to set a breakpoint in that code.
We can get GDB to stop on a library load and hence
try to set the pending breakpoints. See the documentation
for more details relating to this behavior.

If you are not "dlopen"ing libs then leave this unchecked.</string>
          </property>
          <property name="text" >
           <string>Try settings &amp;breakpoints on library loading</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="enableFloatingToolBar_box" >
          <property name="whatsThis" >
           <string>Use the floating toolbar. This toolbar always stays
on top of all windows so that if the app covers KDevelop
you have control of the app through the small toolbar. It
can also be docked into the panel.

This toolbar is in addition to the toolbar in KDevelop.</string>
          </property>
          <property name="text" >
           <string>E&amp;nable floating toolbar</string>
          </property>
          <property name="shortcut" >
           <string>Alt+N</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="dbgTerminal_box" >
          <property name="whatsThis" >
           <string>This allows you to enter terminal input when your
application contains terminal input code (e.g. cin, fgets, etc.).
If you use terminal input in your application then check this option.
Otherwise leave it unchecked.</string>
          </property>
          <property name="text" >
           <string>Enable separate terminal for application &amp;IO</string>
          </property>
          <property name="shortcut" >
           <string>Alt+I</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer>
          <property name="orientation" >
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" >
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </item>
     <item row="0" column="1" >
      <widget class="QGroupBox" name="groupBox_3" >
       <property name="title" >
        <string>Global Output Radix</string>
       </property>
       <layout class="QVBoxLayout" >
        <item>
         <widget class="QRadioButton" name="outputRadixOctal" >
          <property name="text" >
           <string>Oct&amp;al</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="outputRadixHexadecimal" >
          <property name="text" >
           <string>He&amp;xadecimal</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="outputRadixDecimal" >
          <property name="text" >
           <string>Decimal</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item row="1" column="1" >
      <widget class="QGroupBox" name="groupBox_4" >
       <property name="title" >
        <string>Start Debugger With</string>
       </property>
       <layout class="QVBoxLayout" >
        <item>
         <widget class="QRadioButton" name="radioFramestack" >
          <property name="text" >
           <string>Framestack</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="radioGDB" >
          <property name="text" >
           <string>GDB Output</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item row="6" column="0" >
    <widget class="QGroupBox" name="groupBox" >
     <property name="whatsThis" >
      <string>This script is intended for the actual commands needed to connect to a remotely running executable.
	shell sleep 5	wait for remote program to start
	target remote ...	connect to the remote debugger
	continue	[optional] run debugging to the first breakpoint</string>
     </property>
     <property name="title" >
      <string>Remote Debugging</string>
     </property>
     <layout class="QGridLayout" >
      <item row="0" column="0" >
       <widget class="QLabel" name="configGdbScript_label" >
        <property name="text" >
         <string>&amp;Config gdb script:</string>
        </property>
        <property name="wordWrap" >
         <bool>false</bool>
        </property>
        <property name="buddy" >
         <cstring>configGdbScript_edit</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="1" >
       <widget class="KUrlRequester" name="configGdbScript_edit" >
        <property name="toolTip" >
         <string>Gdb configure script</string>
        </property>
        <property name="whatsThis" >
         <string>This script is sourced by gdb when the debugging starts.</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" >
       <widget class="QLabel" name="runShellScript_label" >
        <property name="toolTip" >
         <string/>
        </property>
        <property name="text" >
         <string comment="Shell script to be executed at run time" >R&amp;un shell script:</string>
        </property>
        <property name="wordWrap" >
         <bool>false</bool>
        </property>
        <property name="buddy" >
         <cstring>runShellScript_edit</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="1" >
       <widget class="KUrlRequester" name="runShellScript_edit" >
        <property name="toolTip" >
         <string>Script to start remote application</string>
        </property>
        <property name="whatsThis" >
         <string>This shell script is run after the Config gdb script has been sourced by gdb.
When debugging remotely this script is intended to actually start the remote process.
[It is expected that the debug executable can be reached on the target, maybe by downloading it as a final build step]
1) Find a way to execute a command remotely - rsh, ssh, telnet, ...
2a) Execute "gdbserver ... application" on target.
or if your executable contains the gdb stub
2b) Execute "application" on target.</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0" >
       <widget class="QLabel" name="runGdbScript_label" >
        <property name="text" >
         <string comment="Gdb script to be executed at run time." >Run &amp;gdb script:</string>
        </property>
        <property name="wordWrap" >
         <bool>false</bool>
        </property>
        <property name="buddy" >
         <cstring>runGdbScript_edit</cstring>
        </property>
       </widget>
      </item>
      <item row="2" column="1" >
       <widget class="KUrlRequester" name="runGdbScript_edit" >
        <property name="toolTip" >
         <string>Script to connect with remote application</string>
        </property>
        <property name="whatsThis" >
         <string>This script is sourced by gdb after the two preceding scripts have been executed.
This script is intended for the actual commands needed to connect to a remotely running executable.
	shell sleep 5	wait for remote program to start
	target remote ...	connect to the remote debugger
	continue	[optional] run debugging to the first breakpoint.</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="7" column="0" >
    <spacer>
     <property name="orientation" >
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType" >
      <enum>QSizePolicy::Expanding</enum>
     </property>
     <property name="sizeHint" >
      <size>
       <width>20</width>
       <height>16</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11" />
 <layoutfunction spacing="KDialog::spacingHint" margin="KDialog::marginHint" />
 <pixmapfunction>qPixmapFromMimeSource</pixmapfunction>
 <customwidgets>
  <customwidget>
   <class>KUrlRequester</class>
   <extends>QFrame</extends>
   <header>kurlrequester.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>debuggingShell_edit</tabstop>
  <tabstop>gdbPath_edit</tabstop>
  <tabstop>displayStaticMembers_box</tabstop>
  <tabstop>asmDemangle_box</tabstop>
  <tabstop>breakOnLoadingLibrary_box</tabstop>
  <tabstop>enableFloatingToolBar_box</tabstop>
  <tabstop>dbgTerminal_box</tabstop>
  <tabstop>outputRadixOctal</tabstop>
  <tabstop>outputRadixDecimal</tabstop>
  <tabstop>outputRadixHexadecimal</tabstop>
  <tabstop>configGdbScript_edit</tabstop>
  <tabstop>runShellScript_edit</tabstop>
  <tabstop>runGdbScript_edit</tabstop>
 </tabstops>
 <includes>
  <include location="local" >kurlrequester.h</include>
  <include location="local" >klineedit.h</include>
  <include location="local" >kpushbutton.h</include>
  <include location="local" >kurlrequester.h</include>
  <include location="local" >klineedit.h</include>
  <include location="local" >kpushbutton.h</include>
  <include location="local" >kurlrequester.h</include>
  <include location="local" >klineedit.h</include>
  <include location="local" >kpushbutton.h</include>
  <include location="local" >kurlrequester.h</include>
  <include location="local" >klineedit.h</include>
  <include location="local" >kpushbutton.h</include>
  <include location="local" >kurlrequester.h</include>
  <include location="local" >klineedit.h</include>
  <include location="local" >kpushbutton.h</include>
 </includes>
 <resources/>
 <connections/>
</ui>
